aten/src/ATen/native/SegmentReduce.cpp

```diff
@@ -50,8 +54,12 @@ Tensor _segment_reduce_cpu_kernel(
               initial_value = initial.value().to<scalar_t>();
             } else if (reduction == SegmentReductionType::MAX) {
               initial_value = std::numeric_limits<scalar_t>::lowest();
-            } else if (reduction == SegmentReductionType::MEAN) {
+            } else if (
+                reduction == SegmentReductionType::MEAN ||
+                reduction == SegmentReductionType::SUM) {
               initial_value = 0;
+            } else if (reduction == SegmentReductionType::MIN) {
+              initial_value = std::numeric_limits<scalar_t>::max();
```

Comment:
This should be numeric_limits::infinity (and -infinity for reduction type max), this should work now that you dispatch to floating types only.